sql-plugin/src/main/scala/org/apache/spark/sql/rapids/AggregateFunctions.scala

```diff
@@ -2044,8 +2045,199 @@ class GpuToCpuCollectBufferConverter extends GpuToCpuAggregateBufferConverter {
   }
 }
 
-case class GpuToCpuCollectBufferTransition(
-    override val child: Expression) extends GpuToCpuBufferTransition {
+case class GpuToCpuCollectBufferTransition(override val child: Expression)
+  extends GpuToCpuBufferTransition {
+
+  private lazy val projection = UnsafeProjection.create(Array(child.dataType))
+
+  override protected def nullSafeEval(input: Any): Array[Byte] = {
+    // Converts UnSafeArrayData into binary buffer, according to the serialize method of Collect.
+    // The binary buffer is the binary view of a UnsafeRow, which only contains single field
+    // with ArrayType of elementType. As Collect.serialize, we create an UnsafeProjection to
+    // transform ArrayData to binary view of the single field UnsafeRow. Unlike Collect.serialize,
+    // we don't have to build ArrayData from on-heap array, since the input is already formatted
+    // in ArrayData(UnsafeArrayData).
+    val arrayData = input.asInstanceOf[ArrayData]
+    projection.apply(InternalRow.apply(arrayData)).getBytes
+  }
+}
+
+case class GpuIdentity(child: Expression) extends GpuUnaryExpression {
+  override def prettyName: String = "identity"
+
+  override def dataType: DataType = child.dataType
+
+  override def doColumnar(input: GpuColumnVector): ColumnVector = {
+    TableDebug.get().debug("Final histogram", input.getBase);
+    input.getBase.incRefCount()
+  }
+
+  override def nullable: Boolean = child.nullable
+
+//      override def children: Seq[Expression] = Seq(child)
+
+  //override def canEqual(that: Any): Boolean = true
+
+}
+
+/**
+ * Compute percentile of the input number(s).
+ *
+ * The two 'offset' parameters are not used by GPU version, but are here for the compatibility
+ * with the CPU version and automated checks.
+ */
+abstract class GpuPercentile(childExprs: Seq[Expression], isReduction: Boolean)
+  extends GpuAggregateFunction with Serializable {
+  protected class CudfHistogram(override val dataType: DataType) extends CudfAggregate {
+    override lazy val reductionAggregate: cudf.ColumnVector => cudf.Scalar =
+      (col: cudf.ColumnVector) => col.reduce(ReductionAggregation.histogram(), DType.LIST)
+    override lazy val groupByAggregate: GroupByAggregation = GroupByAggregation.histogram()
+    override val name: String = "CudfHistogram"
+  }
+
+  protected class CudfMergeHistogram(override val dataType: DataType) extends CudfAggregate {
```

Comment:
Same here. Probably should be a top level class or under a GpuPercentile object.

Reply:
> or under a GpuPercentile object.

Oh I didn't know that. Will do.